YAKL_alloc_free.h

```diff
@@ -75,26 +75,28 @@ namespace yakl {
         alloc = [] ( size_t bytes ) -> void* {
           if (bytes == 0) return nullptr;
           void *ptr = sycl::malloc_shared(bytes,sycl_default_stream);
+          sycl_default_stream.memset(ptr, 0, bytes).wait();
           check_last_error();
           sycl_default_stream.prefetch(ptr,bytes);
           return ptr;
         };
         dealloc = [] ( void *ptr ) {
-          sycl::free(ptr, sycl_default_stream).wait();
+          sycl::free(ptr, sycl_default_stream);
           check_last_error();
         };
       #else
         alloc = [] ( size_t bytes ) -> void* {
           if (bytes == 0) return nullptr;
           void *ptr = sycl::malloc_device(bytes,sycl_default_stream);
+          sycl_default_stream.memset(ptr, 0, bytes).wait();
           check_last_error();
           return ptr;
         };
         dealloc = [] ( void *ptr ) {
           sycl_default_stream.wait();
           sycl::free(ptr, sycl_default_stream);
-          sycl_default_stream.wait();
           check_last_error();
+          ptr = nullptr;
```

Comment:
Does this do anything? I thought the pointer address isn't modifiable with this function signature.

Reply:
Right, this wasn't needed. removed